crates/swc_ecma_lexer/src/common/lexer/token.rs

```diff
@@ -0,0 +1,569 @@
+    const PRIVATE: Self;
+    const PROTECTED: Self;
+    const PUBLIC: Self;
+    const READONLY: Self;
+    const ARROW: Self;
+    const REQUIRE: Self;
+    const AWAIT: Self;
+    const BREAK: Self;
+    const CONTINUE: Self;
+    const THIS: Self;
+    const SUPER: Self;
+    const WHILE: Self;
+    const DO: Self;
+    const LPAREN: Self;
+    const RPAREN: Self;
+    const LBRACKET: Self;
+    const RBRACKET: Self;
+    const LBRACE: Self;
+    const FINALLY: Self;
+    const CATCH: Self;
+    const SWITCH: Self;
+    const RBRACE: Self;
+    const FUNCTION: Self;
+    const IF: Self;
+    const ELSE: Self;
+    const CLASS: Self;
+    const NEW: Self;
+    const ABSTRACT: Self;
+    const ACCESSOR: Self;
+    const IMPORT: Self;
+    const PLUS: Self;
+    const MINUS: Self;
+    const BANG: Self;
+    const TILDE: Self;
+    const PLUS_PLUS: Self;
+    const MINUS_MINUS: Self;
+    const DELETE: Self;
+    const TYPEOF: Self;
+    const VOID: Self;
+    const EXTENDS: Self;
+    const SEMI: Self;
+    const OF: Self;
+    const KEYOF: Self;
+    const UNIQUE: Self;
+    const INFER: Self;
+    const USING: Self;
+    const WITH: Self;
+    const ASYNC: Self;
+    const CASE: Self;
+    const DEFAULT: Self;
+    const DEBUGGER: Self;
+
+    fn jsx_name(name: &'a str, lexer: &mut Self::Lexer) -> Self;
+    fn is_jsx_name(&self) -> bool;
+    fn take_jsx_name(self, buffer: &mut Self::Buffer) -> Atom;
+
+    fn str(value: Atom, raw: Atom, lexer: &mut Self::Lexer) -> Self;
+    fn is_str(&self) -> bool;
+    fn take_str(self, buffer: &mut Self::Buffer) -> (Atom, Atom);
+
+    fn template(cooked: LexResult<Atom>, raw: Atom, lexer: &mut Self::Lexer) -> Self;
+    fn is_template(&self) -> bool;
+    fn take_template(self, buffer: &mut Self::Buffer) -> (LexResult<Atom>, Atom);
+
+    fn jsx_text(value: Atom, raw: Atom, lexer: &mut Self::Lexer) -> Self;
+    fn is_jsx_text(&self) -> bool;
+    fn take_jsx_text(self, buffer: &mut Self::Buffer) -> (Atom, Atom);
+
+    fn regexp(content: Atom, flags: Atom, lexer: &mut Self::Lexer) -> Self;
+    fn is_regexp(&self) -> bool;
+    fn take_regexp(self, buffer: &mut Self::Buffer) -> (Atom, Atom);
+
+    fn num(value: f64, raw: Atom, lexer: &mut Self::Lexer) -> Self;
+    fn is_num(&self) -> bool;
+    fn take_num(self, buffer: &mut Self::Buffer) -> (f64, Atom);
+
+    fn bigint(value: Box<BigInt>, raw: Atom, lexer: &mut Self::Lexer) -> Self;
+    fn is_bigint(&self) -> bool;
+    fn take_bigint(self, buffer: &mut Self::Buffer) -> (Box<BigInt>, Atom);
+
+    fn unknown_ident(value: Atom, lexer: &mut Self::Lexer) -> Self;
+    fn is_unknown_ident(&self) -> bool;
+    fn take_unknown_ident(self, buffer: &mut Self::Buffer) -> Atom;
+    fn take_unknown_ident_ref<'b>(&'b self, buffer: &'b mut Self::Buffer) -> &'b Atom;
+
+    fn is_known_ident(&self) -> bool;
+    fn take_known_ident(&self) -> Atom;
+
+    fn starts_expr(&self) -> bool;
+    fn to_string(&self, buffer: &Self::Buffer) -> String;
```

Comment:
This does not seem to be used

Reply:
see: https://github.com/swc-project/swc/pull/10399/files#diff-d22806d12c2473292860573463bef072dd0c36c2612083e03e455645de5c4b76R132